src/app/pages/product/product-details/product-details.component.html

```diff
@@ -66,7 +66,11 @@ <h1 class="product__label">{{product.title | titlecase}}</h1>
                     <option>Sdc</option>
                 </select>
             </form>
-            <button class="green-button" (click)="user ? handleAddToCart(product) : openDialog()">
+            <button class="green-button" *ngIf="!user || !objectKeys(user).length" (click)="openDialog()">
```

Comment:
Do you have a case, when you will have `!user || objectKeys(user).length` 
It's a bit confusing why do you need this check for no keys in user...

Reply:
I can have: null (after remove user from local storage) or empty object (after the first initialization)